### PR TITLE
Add KogitoInfra capablities in RHPAM Kogito Operator

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

https://issues.redhat.com/browse/BAPL-1970

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by vaibhavjainwiz** [comment]
depends on https://github.com/kiegroup/kogito-operator/pull/953

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
@vaibhavjainwiz is this one ready to merge?

**Comment by vaibhavjainwiz** [comment]
It is under review.

File: bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -18,6 +18,16 @@ metadata:
...
+        {
+          "apiVersion": "rhpam.kiegroup.org/v1",
+          "kind": "KogitoInfra",
+          "metadata": {
+            "name": "kogitoinfra-sample"
+          },
+          "spec": {
+            "foo": "bar"
+          }
+        },
```
Comment (sutaakar) [inline bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml:30]:
```suggestion
        {
          "apiVersion": "app.kiegroup.org/v1beta1",
          "kind": "KogitoInfra",
          "metadata": {
            "name": "kogito-infinispan-infra"
          },
          "spec": {
            "resource": {
              "apiVersion": "kafka.strimzi.io/v1beta2",
              "kind": "Kafka",
              "name": "kogito-kafka"
            }
          }
        },
```

File: config/samples/rhpam_v1_kogitoinfra.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: rhpam.kiegroup.org/v1
+kind: KogitoInfra
+metadata:
+  name: kogitoinfra-sample
+spec:
+  # Add fields here
+  foo: bar
```
Comment (sutaakar) [inline config/samples/rhpam_v1_kogitoinfra.yaml:7]:
```suggestion
metadata:
  name: kogito-kafka-infra
spec:
  resource:
    apiVersion: kafka.strimzi.io/v1beta2
    kind: Kafka
    name: kogito-kafka
```

File: controllers/kogitoinfra_controller.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+// +kubebuilder:rbac:groups=apps,resources=deployments;replicasets,verbs=get;create;list;watch;create;delete;update
+// +kubebuilder:rbac:groups=infinispan.org,resources=infinispans,verbs=get;create;list;delete;watch
+// +kubebuilder:rbac:groups=kafka.strimzi.io,resources=kafkas;kafkatopics,verbs=get;create;list;delete;watch
+// +kubebuilder:rbac:groups=keycloak.org,resources=keycloaks,verbs=get;create;list;delete;watch
+// +kubebuilder:rbac:groups=apps,resources=deployments/finalizers,verbs=update
+// +kubebuilder:rbac:groups=eventing.knative.dev,resources=brokers,verbs=get;list;watch
+// +kubebuilder:rbac:groups=eventing.knative.dev,resources=triggers,verbs=get;list;watch;create;delete;update
+// +kubebuilder:rbac:groups=sources.knative.dev,resources=sinkbindings,verbs=get;list;watch;create;delete;update
+// +kubebuilder:rbac:groups=integreatly.org,resources=grafanadashboards,verbs=get;create;list;watch;create;delete;update
+// +kubebuilder:rbac:groups=mongodb.com,resources=mongodb,verbs=get;create;list;watch;delete
```
Comment (sutaakar) [inline controllers/kogitoinfra_controller.go:54]:
```suggestion
// +kubebuilder:rbac:groups=kafka.strimzi.io,resources=kafkas;kafkatopics,verbs=get;create;list;delete;watch
// +kubebuilder:rbac:groups=apps,resources=deployments/finalizers,verbs=update
// +kubebuilder:rbac:groups=integreatly.org,resources=grafanadashboards,verbs=get;create;list;watch;create;delete;update
```

File: controllers/kogitoruntime_finalizer.go [hunk 4]
```diff
@@ -31,6 +32,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+// newKogitoRuntimeFinalizerReconciler returns a new reconcile.Reconciler
```
Comment (sutaakar) [inline controllers/kogitoruntime_finalizer.go:35]: Can be removed?

File: internal/kogitoinfra.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// NewKogitoInfraHandler ...
+func NewKogitoInfraHandler(context operator.Context) manager.KogitoInfraHandler {
+	return &kogitoInfraHandler{context}
+}
+
+// NewNoOpKogitoInfraHandler returns a new instance of manager.KogitoInfraHandler which won't try to interact with the api.KogitoInfraInterface object
+func NewNoOpKogitoInfraHandler(context operator.Context) manager.KogitoInfraHandler {
```
Comment (sutaakar) [inline internal/kogitoinfra.go:32]: kogitoInfraHandlerNoOp seems unused, can it be removed?

**Comment by spolti** [comment]
@vaibhavjainwiz we just merged a fix for the OLM tests, can you please rebase to see if the test will pass?

**Review by spolti — APPROVED** [review]

File: bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (vaibhavjainwiz) [reply bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml:30]: done

File: config/samples/rhpam_v1_kogitoinfra.yaml (hunk 2 above)
Comment (vaibhavjainwiz) [reply config/samples/rhpam_v1_kogitoinfra.yaml:7]: done

File: controllers/kogitoinfra_controller.go (hunk 3 above)
Comment (vaibhavjainwiz) [reply controllers/kogitoinfra_controller.go:54]: done

File: internal/kogitoinfra.go (hunk 5 above)
Comment (vaibhavjainwiz) [reply internal/kogitoinfra.go:32]: yes, its unused. I had removed it.

File: controllers/kogitoruntime_finalizer.go (hunk 4 above)
Comment (vaibhavjainwiz) [reply controllers/kogitoruntime_finalizer.go:35]: done

**Pull request merged**